src/tests/system/tests/test_default_debug_level.py

```diff
@@ -0,0 +1,191 @@
+    assert client.auth.ssh.password("user1", "Secret123"), "Authentication failed"
+    assert start_log_length == log_length(), "Debug messages were generated"
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.Client)
+def test_default_debug_level__fatal_and_critical_failures(client: Client):
+    """
+    :title: Check that messages with levels 0 and 1 are logged for fatal or critical failures
+    :setup:
+        1. Start SSSD with default debug level (config file is created)
+        2. Restrict sssd.conf permissions
+    :steps:
+        1. Restart sssd
+        2. Check logs
+    :expectedresults:
+        1. SSSD failed to start
+        2. Fatal failures (level 0) and Critical failures (level 1) are in log file
+    :customerscenario: True
+    """
+    client.sssd.common.local()
+    client.sssd.default_domain = "local"
+    client.sssd.start(debug_level=None)
+    client.fs.chmod(mode="444", path="/etc/sssd/sssd.conf")
+
+    assert (
+        client.sssd.restart(debug_level=None, raise_on_error=False, apply_config=False).rc != 0
+    ), "SSSD started successfully, which is not expected"
+
+    log_str = client.fs.read("/var/log/sssd/sssd.log")
+    assert "0x0010" in log_str, "Fatal failures were not found in log"
+    assert "0x0020" in log_str, "Critical failures were not found in log"
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.Client)
+def test_default_debug_level__cannot_load_sssd_config(client: Client):
+    """
+    :title: Check that messages with level 2 are logged when SSSD can't load config
+    :setup:
+        1. Set 'domains' to 'non_existing_domain' in sssd section
+    :steps:
+        1. Try to start SSSD with default debug level
+        2. Check logs
+    :expectedresults:
+        1. SSSD failed to start
+        2. Correct error message is in log file
+    :customerscenario: True
+    """
+    client.sssd.sssd["domains"] = "non_existing_domain"
+    assert (
+        client.sssd.start(debug_level=None, raise_on_error=False).rc != 0
+    ), "SSSD started successfully, which is not expected"
+    assert "SSSD couldn't load the configuration database" in client.fs.read("/var/log/sssd/sssd.log")
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_default_debug_level__nonexisting_ldap_server(client: Client):
+    """
+    :title: Check that messages with level 2 are logged when LDAP server doesn't exist
+    :setup:
+        1. Start sssd with default debug level
+        2. Set ldap_uri to a non-existing ldap-server
+    :steps:
+        1. Check logs
+        2. Check default domain status
+    :expectedresults:
+        1. Domain logs should contain a log related to 'going offline'
+        2. LDAP is not connected
+    :customerscenario: True
+    """
+    client.sssd.domain["ldap_uri"] = "ldap://typo.invalid"
+    client.sssd.start(debug_level=None, raise_on_error=False)
+
+    time.sleep(10)
```

Comment:
Remove. As you can see, the test still fails in PR CI.

IFP responder can not start, apparently because:
```
(2023-10-30 11:39:53): [ifp] [sss_iface_connect_address] (0x0020): check_file failed for [/var/lib/sss/pipes/private/sbus-dp_test].
********************** PREVIOUS MESSAGE WAS TRIGGERED BY THE FOLLOWING BACKTRACE:
   *  [ifp] [server_setup] (0x0080): Failed setting process group: Operation not permitted[1]. We might leak processes in case of failure
   *  [ifp] [ldb] (0x0400): server_sort:Unable to register control with rootdse!
   *  (2023-10-30 11:39:53): [ifp] [server_setup] (0x0400): CONFDB: /var/lib/sss/db/config.ldb
   *  (2023-10-30 11:39:53): [ifp] [schedule_responder_idle_timer] (0x2000): Re-scheduling the idle timeout [responder_idle_timeout] for the responder [0x55f7766ebf20]
   *  (2023-10-30 11:39:53): [ifp] [setup_responder_idle_timer] (0x2000): Setting up the idle timeout [responder_idle_timeout] for the responder [0x55f7766ebf20]
   *  (2023-10-30 11:39:53): [ifp] [confdb_init_domain_provider_and_enum] (0x0400): No enumeration for [test]
   *  (2023-10-30 11:39:53): [ifp] [confdb_init_domain_pwd_expire] (0x1000): pwd_expiration_warning is -1
   *  (2023-10-30 11:39:53): [ifp] [sss_get_etc_shells] (0x0400): Found shell /bin/sh in /etc/shells
   *  (2023-10-30 11:39:53): [ifp] [sss_get_etc_shells] (0x0400): Found shell /bin/bash in /etc/shells
   *  (2023-10-30 11:39:53): [ifp] [sss_get_etc_shells] (0x0400): Found shell /usr/bin/sh in /etc/shells
   *  (2023-10-30 11:39:53): [ifp] [sss_get_etc_shells] (0x0400): Found shell /usr/bin/bash in /etc/shells
   *  (2023-10-30 11:39:53): [ifp] [sss_get_etc_shells] (0x0400): Found shell /usr/bin/tmux in /etc/shells
   *  (2023-10-30 11:39:53): [ifp] [sss_get_etc_shells] (0x0400): Found shell /bin/tmux in /etc/shells
   *  (2023-10-30 11:39:53): [ifp] [sss_names_init_from_args] (0x0100): Using re [^((?P<name>.+)@(?P<domain>[^@]+)|(?P<name>[^@]+))$].
   *  (2023-10-30 11:39:53): [ifp] [sss_fqnames_init] (0x0100): Using fq format [%1$s@%2$s].
   *  (2023-10-30 11:39:53): [ifp] [perform_checks] (0x1000): File '/var/lib/sss/pipes/private/sbus-dp_test' must be owned by uid [991].
   *  (2023-10-30 11:39:53): [ifp] [sss_iface_connect_address] (0x0020): check_file failed for [/var/lib/sss/pipes/private/sbus-dp_test].
```

CentOS is built with sssd user, but it looks like the socket is owned by root. This apparently causes problem @alexey-tikhonov do you remember any patch that could cause this? It looks like a regression. It is probably caused by 41427f95751ad087c6a877b86722f8021d01eb39 but there is a conflict if I try to revert it.

It can be fixed by enabling ifp explicitly with `client.sssd.enable_responder("ifp")` instead of relying on dbus activation.

Reply:
Thanks Pavel for looking into this. I applied your suggestion, lets see the results.

Reply:
Hi.

I wouldn't say this is a regression. This is rather a general SSSD problem that RHEL defaults wrt "monitor activation" vs "socket activation" do not match.

RHEL (and CS) by default has 'sssd.conf::user == root' but systemd service descriptions have 'User=sssd'
This doesn't work out of the box for 'autofs', 'pac', 'pam', 'ssh', 'sudo' responders: being run under 'sssd' they can't connect to root owned provider's SBUS socket (or to monitor's SBUS socket after recent update).
Previously IFP 'User=' was hardcoded to root, now it is also '@SSSD_USER@', so it also doesn't work in mixed setups "out of the box". I'll hope we'll fix this mismatch by 2.10 release.

Options at the moment are:
 - run 'ifp' via 'monitor'/'sssd.conf'
 - configure 'User=root' in systemd service file explicitly
 - configure 'user=sssd' in 'sssd.conf' on CS only (but this will probably blow up in different place :) )

Reply:
Ok, thank you for explanation. This is now resolved by adding ifp into enabled services in sssd.conf.